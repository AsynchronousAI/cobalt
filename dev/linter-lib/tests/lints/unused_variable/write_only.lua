local bad1 = {}
bad1.x = 1
bad1.y = 2
bad1["z"] = 1

local badButNotYetImplemented = { x = {} }
badButNotYetImplemented.x.y = 1

local okay1 = ok()
okay1.x = 1
okay1.y = 2

local okay2 = { x = ok() }
okay2.x.y = 1

local okay3 = {}
okay3.x().y = 1
