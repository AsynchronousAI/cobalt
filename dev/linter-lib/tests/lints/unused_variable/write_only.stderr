error[unused_variable]: bad1 is assigned a value, but never used
  ┌─ write_only.lua:1:7
  │
1 │ local bad1 = {}
  │       ^^^^
2 │ bad1.x = 1
  │ ---- `bad1` is only getting written to
3 │ bad1.y = 2
  │ ---- `bad1` is only getting written to
4 │ bad1["z"] = 1
  │ ---- `bad1` is only getting written to

